test/java/com/rapleaf/jack/TestGenericQuery.java

```diff
@@ -0,0 +1,159 @@
+package com.rapleaf.jack;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.rapleaf.jack.queries.GenericQuery;
+import com.rapleaf.jack.queries.QueryEntry;
+import com.rapleaf.jack.queries.QueryOrder;
+import com.rapleaf.jack.test_project.DatabasesImpl;
+import com.rapleaf.jack.test_project.IDatabases;
+import com.rapleaf.jack.test_project.database_1.iface.ICommentPersistence;
+import com.rapleaf.jack.test_project.database_1.iface.IPostPersistence;
+import com.rapleaf.jack.test_project.database_1.iface.IUserPersistence;
+import com.rapleaf.jack.test_project.database_1.models.Comment;
+import com.rapleaf.jack.test_project.database_1.models.Post;
+import com.rapleaf.jack.test_project.database_1.models.User;
+
+import static org.junit.Assert.*;
+
+public class TestGenericQuery {
```

Comment:
As part of making some of the changes discussed above you should add tests for those specific cases so it's clear what our intention was / we make sure they work.